src/progress.jl

```diff
@@ -0,0 +1,110 @@
+import Base: done
+
+export ProgressBar, progress!, msg!, done, right_text!, @progress
+
+type ProgressBar
+  leftText::String
+  rightText::String
```

Comment:
I'll make the switch to not having any state in the julia object, seems a lot better.